server/src/main/java/org/opensearch/search/backpressure/trackers/HeapUsageTracker.java

```diff
@@ -34,66 +36,22 @@
  * @opensearch.internal
  */
 public class HeapUsageTracker extends TaskResourceUsageTracker {
-    private static final long HEAP_SIZE_BYTES = JvmStats.jvmStats().getMem().getHeapMax().getBytes();
-
-    private static class Defaults {
-        private static final double HEAP_PERCENT_THRESHOLD = 0.005;
-        private static final double HEAP_VARIANCE_THRESHOLD = 2.0;
-        private static final int HEAP_MOVING_AVERAGE_WINDOW_SIZE = 100;
-    }
-
-    /**
-     * Defines the heap usage threshold (in percentage) for an individual task before it is considered for cancellation.
-     */
-    private volatile double heapPercentThreshold;
-    public static final Setting<Double> SETTING_HEAP_PERCENT_THRESHOLD = Setting.doubleSetting(
-        "search_backpressure.search_shard_task.heap_percent_threshold",
-        Defaults.HEAP_PERCENT_THRESHOLD,
-        0.0,
-        1.0,
-        Setting.Property.Dynamic,
-        Setting.Property.NodeScope
-    );
-
-    /**
-     * Defines the heap usage variance for an individual task before it is considered for cancellation.
-     * A task is considered for cancellation when taskHeapUsage is greater than or equal to heapUsageMovingAverage * variance.
-     */
-    private volatile double heapVarianceThreshold;
-    public static final Setting<Double> SETTING_HEAP_VARIANCE_THRESHOLD = Setting.doubleSetting(
-        "search_backpressure.search_shard_task.heap_variance",
-        Defaults.HEAP_VARIANCE_THRESHOLD,
-        0.0,
-        Setting.Property.Dynamic,
-        Setting.Property.NodeScope
-    );
-
-    /**
-     * Defines the window size to calculate the moving average of heap usage of completed tasks.
-     */
-    private volatile int heapMovingAverageWindowSize;
-    public static final Setting<Integer> SETTING_HEAP_MOVING_AVERAGE_WINDOW_SIZE = Setting.intSetting(
-        "search_backpressure.search_shard_task.heap_moving_average_window_size",
-        Defaults.HEAP_MOVING_AVERAGE_WINDOW_SIZE,
-        0,
-        Setting.Property.Dynamic,
-        Setting.Property.NodeScope
-    );
-
+    public static final long HEAP_SIZE_BYTES = JvmStats.jvmStats().getMem().getHeapMax().getBytes();
```

Comment:
The `HEAP_SIZE_BYTES` has to go back to `private` please

Reply:
We need the access of `HEAP_SIZE_BYTES` in `Search[Shard]TaskSettings` for `getHeapBytesThreshold` since that should not be moved to `HeapUsageTracker`. I will access the value using a public method `HeapUsageTracker.getHeapSizeBytes`. Hope that's fine.

Reply:
You could always calculate the threshold based on `getHeapThreshold` (this is the only setting  you need), and `HeapUsageTracker` know` how calculate the threshold in bytes